### DISCUSSION TITLE
Do clickhouse open-source supports role-based access for s3?

### DISCUSSION THREAD

**Discussion by purush7** [ClickHouse/ClickHouse#52220] [Q&A] [answered]
Do clickhouse (not cloud) currently supports role-based access for s3?

I have seen in clickhouse-cloud docs that it supports role-based by using `extra_credentials` and the syntax for s3 function in docs says
`s3(path [, NOSIGN | aws_access_key_id, aws_secret_access_key] [,format] [,structure] [,compression])`.

Is there a way to use role-based for s3 function? BTW I want to use s3 as function not as a table.

Thanks in advance

**Comment by alexey-milovidov** [comment, marked as answer]
No, it is only in the Cloud.

**Reply by aser86** [reply]
Are there plans to provide the function in the not cloud version?
